wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/OverridePropertiesApplier.java

```diff
@@ -90,6 +91,10 @@ public JsonNode applyPropertiesOverride(
     return devfileNode;
   }
 
+  private String[] parseSegments(String key) {
+    return key.split("\\.", key.startsWith("attributes.") ? 2 : 0);
```

Comment:
can you please write a comment here why it's doing what it's doing? I do understand it now, but I'm not sure in few months... It is definitely not a straight-forward and easy to read code.

Reply:
or maybe use `if startsWith("attributes") -> key.replace("attributes.", "")` ? Would it be more readable?

Reply:
added comment